File: Data Structures & Algorithms/max-water-container/submission-1.java
class Solution {
    public int maxArea(int[] heights) {
        int n=heights.length;

        

        int i=0,j=n-1;
        int res=-1;

        while(i<j)
        {
            int width=j-i;
            int height=Math.min(heights[i],heights[j]);
            res=Math.max(res,height*width);
            if(heights[i]<=heights[j]) i++;
            else
                j--;
        }
        return res;
    }
}

File: Data Structures & Algorithms/eating-bananas/submission-1.py
class Solution:
    def minEatingSpeed(self, piles: List[int], h: int) -> int:


        max_ele = max(piles)


        min_ele = 0

        while min_ele < max_ele :

            mid = (min_ele + max_ele)//2

            res = 0
            
            for p in piles:
                if mid == 0:
                    break
                res += math.ceil(p/mid)

            if res > h or mid == 0:

                min_ele = mid + 1
            
            else:

                max_ele = mid 

        return max_ele

